pkg/operator/reconcile.go

```diff
@@ -33,29 +33,33 @@ import (
 // but also
 //   sync random secret between oauthclient & oauthclient-secret
 //   sync route.host between route, oauthclient.redirectURIs & configmap.baseAddress
-func ReconcileConsole(cr *v1alpha1.Console) {
-
-	CreateServiceIfNotPresent(cr)
-	rt, _ := CreateRouteIfNotPresent(cr)
-	CreateConsoleConfigMapIfNotPresent(cr, rt)
-	CreateOauthClientIfNotPresent(cr, rt)
-	CreateConsoleDeploymentIfNotPresent(cr)
-
-	UpdateOauthClientIfNotInSync(cr, rt)
-
-	//rt, _ := CreateRoute(cr)
-	//
-	//// fetching the route to get it with a host annotation
-	//_ = sdk.Get(rt)
-	//
-	//CreateConsoleConfigMap(cr, rt)
-	//CreateOAuthClient(cr, rt)
-	//
-	//CreateConsoleDeployment(cr)
-	//
-	//// ensure these stay in sync.
-	//// can probably dedupe some work here
-	// UpdateOauthClient(cr, rt)
-	//
-	//// this should prob return errors :)
+func ReconcileConsole(cr *v1alpha1.Console) error {
+
+	_, err := CreateServiceIfNotPresent(cr)
```

Comment:
I think order matters in two cases:

1. We need the route hostname before we can update the OAuth client redirect URL.
2. We should create the config map and OAuth client secret before the deployment since those are only read on start up.

I haven't looked closely at what you have for reconcile, but if the config map or OAuth secret changes, you'll need to delete the existing console pods (or force a rollout). The console isn't watching the filesystem for config changes.